File: Client/client-app/src/services/AlumnoService.ts
import { Alumno, AlumnoVM, Messages } from "@/utilities/ViewModels";
import axios from "axios";

export default {
    // async $LoggedUser(numeroControl: number, password: string) {
    //     return axios.get<Messages.ResponseBase<Alumno>>(`https://sitecglobal.azurewebsites.net/api/Alumno/Login?numeroControl=${numeroControl}&password=${password}`);
    // },
    async $GetAlumnoByAlumnoId(numeroControl: number) {
        return axios.get<Messages.ResponseBase<AlumnoVM>>(`https://localhost:7023/api/alumno/GetAlumnoByNumeroDeControl?numeroControl=${numeroControl}`);
    },
    async $LoggedUser(numeroControl: number, password: string) {
        return axios.get<Messages.ResponseBase<Alumno>>(`https://localhost:7023/api/alumno/Login?numeroControl=${numeroControl}&password=${password}`);
    },
};

File: BackEnd/SITEC.GLOBAL/BussinessLogic/Services/AlumnoService.cs
using Microsoft.AspNetCore.Http.HttpResults;
using SITEC.GLOBAL.BussinessLogic.Messages;
using SITEC.GLOBAL.BussinessLogic.ViewModels;
using SITEC.GLOBAL.Data.Repository;
using SITEC.GLOBAL.Domain.Models;

namespace SITEC.GLOBAL.BussinessLogic.Services
{
    public class AlumnoService
    {
        private readonly CustomAlumnoRepository _customAlumnoRepository;
        public AlumnoService(CustomAlumnoRepository customAlumnoRepository)
        {
            _customAlumnoRepository = customAlumnoRepository;
        }
        public ResponseBase<AlumnoVM> GetAlumnoByAlumnoId(int alumnoId)
        {
            ResponseBase<AlumnoVM> result = new();
            result.Message = "Conectado Correctamente";
            Alumno alumno = _customAlumnoRepository.GetAlumnoByAlumnoId(alumnoId);
            AlumnoAdapter adapter = new(alumno);
            AlumnoVM alumnoVM = adapter.ToVM();
            result.Result = alumnoVM;
            result.Type = ResponseType.Success;
            return result;
        }
        public ResponseBase<Alumno> GetLoggedUser(int numeroControl, string password)
        {
            ResponseBase<Alumno> result = new();
            result.Result = _customAlumnoRepository.LoggedUser(numeroControl, password);
            if(result.Result != null) 
            {
                if(result.Result.NumeroControl > 0)
                {
                    result.Type = ResponseType.Success;
                    result.Message = "Usuario Encotrado";
                }
                else
                {
                    result.Type = ResponseType.Warning;
                    result.Message = "Usuario no encontrado";
                }
            }
            else
            {
                result.Type = ResponseType.Error;
                result.Message = "Hubo un problema al intentar loggearte";
            }
            return result;
        }
    }
}
